### ISSUE TITLE
search through all tabs (song / album / artist / etc)

### ISSUE THREAD

**Issue by rmmanseau** [UnicornsOnLSD/finamp#293]
the search feature is currently limited to whichever tab you happen to be browsing. usually this is a bit confusing to me because I expect the search function to show results from all categories.

**Comment by Y0ngg4n** [comment]
Yes that is true but needs a complete rewrite of the search function.

**Comment by jmshrv** [comment]
The design for this could be conceptualised at some point in #220, it wouldn't make sense in the current UI

**Comment by Chaphasilor** [comment]
Most of the code for this is already working as part of the Android Auto logic:

https://github.com/jmshrv/finamp/blob/87a58b7c7420afce407a667d7839d7a4bc602c6b/lib/services/android_auto_helper.dart#L207-L293

It just needs to get integrated into the main application, which requires changes to the UI. A basic "search page" could be implemented independently of other UI changes, so that i.e. the search button opens a new search screen/sheet as a temporary solution, until the new navigation UI is done.
